service/worker/archiver/handler.go

```diff
@@ -167,7 +167,7 @@ func (h *handler) handleHistoryRequest(ctx workflow.Context, request *ArchiveReq
 	localActCtx := workflow.WithLocalActivityOptions(ctx, lao)
 	err = workflow.ExecuteLocalActivity(localActCtx, deleteHistoryActivity, *request).Get(localActCtx, nil)
 	if err != nil {
-		logger.Error("deleting history failed, this means zombie histories are left", tag.Error(err))
+		logger.Error("deleting workflow execution failed all retires, skip workflow deletion", tag.Error(err))
```

Comment:
Will the workflow data stay in Db if it just skip the deletion?

Reply:
Yes. Archival workflow will only retry up to 5min for both uploading and deletion and then give up. This is the limitation of the existing archival design. The issue will be gone once we have a separate archival queue.

For now, user should monitor the metrics for archival delete non-retryable error and use admin wf del command to manually delete those workflows from DB.